src/NonLinearProgram/nlp_utilities.jl

```diff
@@ -0,0 +1,556 @@
+# Copyright (c) 2020: Andrew Rosemberg and contributors
+#
+# Use of this source code is governed by an MIT-style license that can be found
+# in the LICENSE.md file or at https://opensource.org/licenses/MIT.
+
+"""
+    fill_off_diagonal(H)
+
+Filling the off-diagonal elements of a sparse matrix to make it symmetric.
+"""
+function fill_off_diagonal(H)
+    ret = H + H'
+    row_vals = SparseArrays.rowvals(ret)
+    non_zeros = SparseArrays.nonzeros(ret)
+    for col in 1:size(ret, 2)
+        for i in SparseArrays.nzrange(ret, col)
+            if col == row_vals[i]
+                non_zeros[i] /= 2
+            end
+        end
+    end
+    return ret
+end
+
+"""
+    compute_optimal_hessian(evaluator::MOI.Nonlinear.Evaluator, rows::Vector{JuMP.ConstraintRef}, x::Vector{JuMP.VariableRef})
+
+Compute the optimal Hessian of the Lagrangian.
+"""
+function compute_optimal_hessian(
+    model::Model,
+    rows::Vector{MOI.Nonlinear.ConstraintIndex},
+)
+    sense_multiplier = sense_mult(model)
+    evaluator = model.cache.evaluator
+    y = [model.y[model.model.nlp_index_2_constraint[row].value] for row in rows]
```

Comment:
Is there a proper getter to access `model.model.nlp_index_2_constraint`?

Reply:
Not that I know of. @joaquimg , do you ?

Reply:
if there was, it would be a MOI.get...